rmc/utils/constraint.py

```diff
@@ -1239,6 +1239,38 @@ def get_oe_annotation(ht: hl.Table, freeze: int) -> hl.Table:
     return ht.transmute(oe=hl.coalesce(ht.section_oe, ht.transcript_oe))
 
 
+def dedup_annot(
+    ht: hl.Table,
+    annot: str,
+    get_min: bool,
+    key_fields: Tuple[str, str] = ("locus", "alleles"),
+) -> hl.Table:
+    """
+    Deduplicate annotation within input Table.
+
+    Function will group input Table by `key_fields`, select either the min or max value of
+    `annot`, and optionally retain transcript ID.
+
+    :param ht: Input Table.
+    :param annot: Name of annotation to be deduplicated.
+    :param get_min: Whether to get minimum value of annotation.
+    :param keep_transcript: Whether to retain transcript ID.
+    :param key_fields: Key fields to use for deduplication.
+    """
+    ht = ht.select(annot).key_by(*(key_fields))
```

Comment:
Doesn't this already assume HT is keyed by `key_fields`? If not keyed, then `select` would drop them. Hope I'm getting it right

Reply:
yes, good point!